torch/distributions/kl.py

```diff
@@ -163,8 +163,11 @@ def kl_divergence(p, q):
 @register_kl(Bernoulli, Bernoulli)
 def _kl_bernoulli_bernoulli(p, q):
     t1 = p.probs * (p.probs / q.probs).log()
+    t1[(p.probs == 0) | (q.probs == 1)] = 0
```

Comment:
Can you explain why the `(q.probs == 1)` is needed? And why the `p.probs == 1` term below is needed?

Reply:
The term `t2` is `(1 - p.probs) * ln ((1 - q.probs) / (1 - p.probs))`. If `p.probs` == 1, then the log term becomes undefined, but is reduced to 0 by `1 - p.probs`.

Regarding `q.probs == 1`, that was a mistake on my side. Sorry.